.github/workflows/security-scan.yml

```diff
@@ -0,0 +1,356 @@
+---
+name: "Security Scanning"
+
+on:
+  push:
+    branches:
+      - "main"
+      - "release-*"
+  pull_request:
+    branches:
+      - "main"
+      - "release-*"
+  schedule:
+    - cron: '0 12 * * 1'
+
+permissions:
+  contents: read
+  security-events: write
+  actions: read
+
+jobs:
+  codeql:
+    name: "CodeQL SAST Analysis"
+    runs-on: 'ubuntu-22.04'
+    timeout-minutes: 360
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [go]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Setup Go
+        uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5
+        with:
+          go-version: '1.24.1'
+          cache: true
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@39edc492dbe16b1465b0cafca41432d857bdb31a
+        with:
+          languages: ${{ matrix.language }}
+      - name: Autobuild
+        uses: github/codeql-action/autobuild@39edc492dbe16b1465b0cafca41432d857bdb31a
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@39edc492dbe16b1465b0cafca41432d857bdb31a
+        with:
+          category: "/language:${{matrix.language}}"
+          upload: true
+
+  osv-scanner:
+    name: "OSV Vulnerability Scan"
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5
+        with:
+          go-version: '1.24.1'
+          cache: true
+      - name: Generate SBOM
+        run: |
+          go mod tidy
+          go mod download
+          go install github.com/CycloneDX/cyclonedx-gomod/cmd/cyclonedx-gomod@latest
+          if [ -f "go.mod" ]; then
+            echo "Generating SBOM from go.mod..."
+            cyclonedx-gomod mod -output bom.xml || {
+              echo "SBOM generation failed, trying alternative approach..."
+              go list -mod=readonly -m -json all > modules.json || echo "Module listing failed"
+              if [ -f "modules.json" ]; then
+                echo "Generated modules.json as fallback"
+              fi
+            }
+          else
+            echo "No go.mod found, skipping SBOM generation"
+          fi
+      - name: Install OSV Scanner
+        run: |
+          go install github.com/google/osv-scanner/cmd/osv-scanner@latest
+      - name: Get current date for cache key
+        id: date
+        run: echo "date=$(date +'%Y-%m-%d')" >> $GITHUB_OUTPUT
+      - name: Cache OSV database
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cache/osv-scanner
+            ~/.osv-scanner
+          key: osv-db-${{ steps.date.outputs.date }}
+          restore-keys: |
+            osv-db-
```

Comment:
Why is this here? In what scenario does it accomplish something? Does it accomplish something that we really want?

`${{ steps.date.outputs.date }}` will never be the empty string, right?